### ISSUE TITLE
can not validate uuid field

### ISSUE THREAD

**Issue by agrozyme** [gookit/validate#280]
**System (please complete the following information):**

 - OS: `linux`
 - GO Version: `1.23`
 - Pkg Version: `1.5.3`

**Describe the bug**

When struct has uuid field, run validate.New to get panic "reflect: call of reflect.Value.IsNil on array Value"

**To Reproduce**

```go
package main

import (
	"fmt"

	"github.com/google/uuid"
	"github.com/gookit/validate"
)

type Test struct {
	ID uuid.UUID
}

func main() {
	data := Test{}
	_ = validate.New(data).ValidateErr()
	fmt.Println("OK")
}



```

**Expected behavior**

show  "OK"

**Comment by vincentdaniel** [comment]
I see that since [this change](https://github.com/gookit/validate/pull/260), we call `IsNil` in the case of `reflect.Array`: https://github.com/gookit/validate/blob/d0e50a8d41c4e680bdcdb5b0ab1150da29ff91e9/data_source.go#L383
But `IsNil` doesn't have a case for `Array`: https://cs.opensource.google/go/go/+/refs/tags/go1.23.4:src/reflect/value.go;l=1553

**Comment by inhere** [comment]
Fixed. Released the v1.5.4
